Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -3440,3 +3474,34 @@ void webkitWebViewBaseCallAfterNextPresentationUpdate(WebKitWebViewBase* webView
 {
     webkitWebViewBaseNextPresentationUpdateMonitorStart(webViewBase, WTFMove(callback));
 }
+
+#if USE(GTK4)
+void webkitWebViewBaseSetPlugID(WebKitWebViewBase* webViewBase, const String& plugID)
+{
+#if defined(GTK_ACCESSIBILITY_ATSPI)
+    WebKitWebViewBasePrivate* priv = webViewBase->priv;
+
+    if (priv->socketAccessible) {
+        gtk_accessible_set_accessible_parent(priv->socketAccessible.get(), nullptr, nullptr);
+        priv->socketAccessible = nullptr;
+    }
+
+    Vector<String> tokens = plugID.split(':');
+    ASSERT(tokens.len() == 2);
+
+    GUniqueOutPtr<GError> error;
+    GUniquePtr<char> busName(g_strdup_printf(":%s", tokens[0].utf8().data()));
+
+    priv->socketAccessible = adoptGRef(gtk_at_spi_socket_new(busName.get(), tokens[1].utf8().data(), &error.outPtr()));
+
+    if (priv->socketAccessible) {
+        auto* widget = gtk_widget_get_first_child(GTK_WIDGET(webViewBase));
+        gtk_accessible_set_accessible_parent(priv->socketAccessible.get(), GTK_ACCESSIBLE(webViewBase), GTK_ACCESSIBLE(widget));
+    } else
+        g_warning("Error creating socket: %s", error->message);
```

Comment:
A little more info on what is going wrong would be helpful:

```suggestion
        g_warning("Error creating WebKitWebView a11y socket: %s", error->message);
```